pkg/tagset/builder.go

```diff
@@ -0,0 +1,98 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.Datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package tagset
+
+import "github.com/twmb/murmur3"
+
+// Builder is used to build tagsets tag-by-tag, before producing Tags instance
+// when it is closed. Builders are not threadsafe.
+//
+// It is invalid to use a Builder after Close, as it may be re-used by other goroutines.
+//
+// In general, Builders are intended to be used in a single method.  Avoid
+// storing builders in structs.
+type Builder struct {
+	factory Factory
+	tags    []string
+	hashes  []uint64
+	hash    uint64
+	seen    map[uint64]struct{}
+}
+
+// newBuilder creates a new builder. This must be reset()
+// before use.
+func newBuilder(factory Factory) *Builder {
+	return &Builder{
+		factory: factory,
+		tags:    nil,
+	}
```

Comment:
We do always reset when creating a builder:
https://github.com/DataDog/datadog-agent/blob/6b04791007ae0104d5f8068ee7fcfa5de383d3f0/pkg/tagset/base_factory.go#L17-L28
but I think you're asking why this isn't done in the constructor.  I could reorganize it so that the constructor takes a capacity and reset's to that capacity, but then we either double-reset or the NewBuilder function becomes more complex.  I don't think that solution is any more maintainable than what I've got now, but it's not worse.  I'm happy to make that change.

Reply:
Done.